listing all buildtest config profiles list

kingxleyy opened the pull request:
add a command buildtest config profiles list

kingxleyy committed:
Add command menu in argparse

kingxleyy committed:
Resolved conflict to Add command menu in argparse

kingxleyy committed:
Added removed parenthesis

kingxleyy committed:
removed unwanted lines

kingxleyy committed:
Add command menu in argparse,  implementation,entry in buildtest help config and regression test

kingxleyy committed:
fixing errors in previous comments

kingxleyy committed:
fixing conflicts

kingxleyy committed:
re-doing the pull request

kingxleyy committed:
Merge remote-tracking branch 'origin/kingsley_buildtest' into kingsley_buildtest

# Conflicts:
#	buildtest/cli/show.py

kingxleyy committed:
re-doing the pull request

kingxleyy committed:
adding options in bash_completion.sh

Mendi03 commented on buildtest/cli/config.py:
```diff
@@ -153,6 +155,9 @@ def view_configuration(configuration, theme=None, pager=None):
 
+def view_profiles(configuration, theme=None, pager=None):
+    """Display the list of profiles for buildtest configuration file.This implements command ``buildtest config profiles list`
+    """
```
you have no code for this method.

Mendi03 replied:
If this prints the requested information needed from the feature then it is correct. If not, you need to probably add code for this method. Not sure myself.

Mendi03 commented on bash_completion.sh:
```diff
@@ -261,7 +261,9 @@ _buildtest ()
-          local opts="--help --pager --theme -h -t"
+        COMPREPLY=( $( compgen -W "${opts}" -- $cur ) );;
+        profiles)
+          local opts="--help --pager --theme -h -t --yaml"
```
you are modifying the code for `view`. Make sure you leave it as it was.  You did not write `profile` correctly nor put it in the correct lines, this is what is causing errors. Look at the code carefully and write the code syntactically correct.

Mendi03 commented on bash_completion.sh:
```diff
@@ -237,7 +237,7 @@ _buildtest ()
 
     config|cg)
-      local cmds="-h --help co compilers e edit ex executors p path systems validate v view"
+      local cmds="-h --help co compilers e edit ex executors p path systems validate v view profiles"
```
`profiles` should be `profile`. This is what you wrote in buildtest/cli/config.py

kingxleyy committed:
Merge branch 'buildtesters:devel' into kingsley_buildtest

kingxleyy committed:
modifying code for profile, chaning profiles to profile

kingxleyy committed:
Merge remote-tracking branch 'origin/kingsley_buildtest' into kingsley_buildtest

kingxleyy committed:
changing profiles to profile in test_config

kingxleyy committed:
fixing errors in config

kingxleyy committed:
fixing errors in _init_

Mendi03 commented on buildtest/cli/config.py:
```diff
@@ -23,6 +23,9 @@ def config_cmd(args, configuration, editor, system):
 
+    if args.config in ["profile"]:
+        view_profiles(configuration, theme=args.theme, pager=args.pager)
+
```
The name of the method is not the same as the method definition in line 158. Also, I believe you might need to add the yaml argument here.

kingxleyy replied:
fixed

kingxleyy committed:
fixing errors in line 27 of config.py

kingxleyy committed:
resolving conflicts in bash completion

shahzebsiddiqui commented on buildtest/cli/__init__.py:
```diff
@@ -396,7 +398,29 @@ def misc_menu(subparsers):
+            "view",
+            "-v"
+        ],
+    )
```
This does not make any sense. You should expect to see a command name `buildtest config profiles` and this is not doing what you want. If you dont see this command then you have not done this correctly.

kingxleyy committed:
fixing erros

kingxleyy committed:
fixing errors in init_py

kingxleyy committed:
Merge branch 'devel' into kingsley_buildtest

shahzebsiddiqui commented on buildtest/cli/config.py:
```diff
@@ -23,6 +23,9 @@ def config_cmd(args, configuration, editor, system):
 
+    if args.config in ["profile"]:
+        view_profile(configuration, theme=args.theme, pager=args.pager)
+
```
change to `elif`

kingxleyy replied:
done

kingxleyy committed:
fixing errors in config, compliers and init py

kingxleyy committed:
Merge branch 'buildtesters:devel' into kingsley_buildtest

kingxleyy committed:
Merge remote-tracking branch 'origin/kingsley_buildtest' into kingsley_buildtest

Mendi03 commented on buildtest/cli/__init__.py:
```diff
@@ -396,6 +398,23 @@ def misc_menu(subparsers):
+        help="Specify a color theme, Pygments style to use when displaying output. See https://pygments.org/docs/styles/#getting-a-list-of-available-styles for available themese",
+        choices=list(STYLE_MAP.keys()),
+    )
+    return parent_parser
```
Not sure about the purpose of this method. I don't think you need this in the first place because the `profile` command specified in the issue needs to be under the `config_menu` method.

Mendi03 commented on buildtest/cli/compilers.py:
according to the issue, I don't think you needed to modify this file whatsoever.

Mendi03 commented on buildtest/cli/config.py:
```diff
@@ -23,6 +23,9 @@ def config_cmd(args, configuration, editor, system):
 
+    elif args.config in ["profile"]:
+        view_profile(configuration, theme=args.theme, pager=args.pager)
+
```
Here it is saying that the `view_profile` method should be called when someone writes the `profile` argument in the terminal. However, you have not written any of this logic in the __init__.py file. As I stated on a previous comment, you need to add this somewhere in the config_menu method.

kingxleyy committed:
fixing errors in config, init py

Mendi03 commented on buildtest/cli/__init__.py:
```diff
@@ -975,7 +994,9 @@ def config_menu(subparsers, parent_parser):
-
+    profile = subparsers_config.add_parser(
+        "profile", aliases=["profile"], help="Search profile"
+    )
```
This looks right, you might need to add an argument `list` to this subparser as specified in the issue though.

kingxleyy replied:
done

kingxleyy committed:
adding arguments

shahzebsiddiqui commented on buildtest/cli/__init__.py:
```diff
@@ -396,6 +398,23 @@ def misc_menu(subparsers):
+    parent_parser["theme"] = argparse.ArgumentParser(add_help=False)
+    parent_parser["theme"].add_argument(
+        "-t",
+        compliers="buildtest config profile",
```
why do you have this line, please remove `compliers="buildtest config profile",`

kingxleyy replied:
done

Mendi03 commented on buildtest/cli/__init__.py:
```diff
@@ -1063,6 +1084,10 @@ def config_menu(subparsers, parent_parser):
 
+    # buildtest config profile
+    profile.add_argument(
+        "-y", "--yaml", "--theme","view", help="List Profile details in YAML Format"
```
This is not right, you need a `list` argument, not a list of arguments.

shahzebsiddiqui commented on buildtest/cli/__init__.py:
```diff
@@ -1063,6 +1084,10 @@ def config_menu(subparsers, parent_parser):
+    # buildtest config profile
+    profile.add_argument(
+        "-y", "--yaml", "--theme","view", help="List Profile details in YAML Format"
+    )
```
this is not correct. You can't have `--theme` and `view`.

shahzebsiddiqui commented on buildtest/cli/config.py:
```diff
@@ -153,6 +155,11 @@ def view_configuration(configuration, theme=None, pager=None):
+    """Display the list of profile for buildtest configuration file.This implements command ``buildtest config profiles list`
+    """
+
+    console.print(configuration.file)
```
this is not right for the implementation. You need to index the dictionary and list all the profiles. the `configuration.file` is the full path to configuration file.

shahzebsiddiqui committed:
fix issue with argparse and all the bugs related to building the parser

shahzebsiddiqui committed:
fix bash completion script with the appropriate commands

shahzebsiddiqui committed:
add --theme option for 'buildtest config profiles'
remove method from buildtest/cli/compilers.py this was not suppose to be added here

shahzebsiddiqui committed:
add command 'buildtest config profiles list' and add the options --theme, --yaml.
add implementation for listing profiles as raw text and yaml format that supports
color themes specified by --theme option.
we raise an exception when no themese are specified

shahzebsiddiqui committed:
fix bash completion script to add support for 'buildtest config profile list' and its options

shahzebsiddiqui committed:
refactored the method and renamed it to list_profiles. This method can be called
directly and added method variables theme, print_yaml to configure how profiles can be displayed.
Add regression test for listing profiles and in yaml format

shahzebsiddiqui committed:
running isort and black since code wasnt formatted properly

shahzebsiddiqui commented on buildtest/settings/config.yml:
```diff
@@ -1,45 +1,23 @@
-    # specify list of hostnames where buildtest can run for given system record
-    hostnames: [".*"]
-
-    # system description
```
@kingxleyy this was probably my mistake. Please revert this change back to original file shown in `devel` branch. Just copy the entire file and commit it locally and push it so there are no changes shown for this file

kingxleyy replied:
done

kingxleyy committed:
adding table in show.py

kingxleyy committed:
fixing error config yml

kingxleyy committed:
documentation

kingxleyy committed:
documentation

kingxleyy committed:
Merge branch 'buildtesters:devel' into kingsley_buildtest

shahzebsiddiqui commented on bash_completion.sh:
```diff
@@ -269,6 +272,19 @@ _buildtest ()
+            COMPREPLY=( $( compgen -W "${opts}" -- $cur ) )
+          fi
+
+          case "${prev}" in --theme|-t)
```
can you remove the `-t` option this will be removed in https://github.com/buildtesters/buildtest/pull/1576

shahzebsiddiqui commented on bash_completion.sh:
```diff
@@ -269,6 +272,19 @@ _buildtest ()
+          local opts="--help -h list"
+          COMPREPLY=( $( compgen -W "${opts}" -- $cur ) )
+          if [[ "${prev}" == "list" ]]; then
+            local opts="--help  --theme --yaml -h -t -y"
```
same here remove the `-t`

kingxleyy replied:
done

shahzebsiddiqui commented on buildtest/cli/config.py:
```diff
@@ -154,6 +159,31 @@ def view_configuration(configuration, theme=None, pager=None):
 
 
+def list_profiles(configuration, theme=None, print_yaml=None):
+    """Display the list of profile for buildtest configuration file.This implements command ``buildtest config profiles list``
```
add a space after the period `file. This`

kingxleyy replied:
done

shahzebsiddiqui commented on buildtest/settings/config.yml:
```diff
@@ -1,45 +1,23 @@
-    # specify list of hostnames where buildtest can run for given system record
-    hostnames: [".*"]
-
-    # system description
```
please dont commit `buildtest/settings/config.yml` this was a result of running the regression test which autoupdates the file. Don't do `git add .` or something. Just copy the content from `devel` branch and paste it in that should avoid showing this commit diff for this file

kingxleyy replied:
noted

shahzebsiddiqui replied:
@kingxleyy i still see this file ``buildtest/settings/config.yml` in commit diff

shahzebsiddiqui commented on buildtest/cli/__init__.py:
```diff
@@ -988,6 +988,24 @@ def config_menu(subparsers, parent_parser):
     )
+    # buildtest config profile
+    profile = subparsers_config.add_parser(
+        "profile", help="Query profile from buildtest configuration"
```
can you change the command to `profiles` with an `s`. So do the following

```python
    profile = subparsers_config.add_parser(
        "profiles", help="Query profile from buildtest configuration")
```

Note the `s` is for the argument in add_parser not the variable `profile`.

kingxleyy replied:
done

shahzebsiddiqui commented on buildtest/cli/__init__.py:
```diff
@@ -988,6 +988,24 @@ def config_menu(subparsers, parent_parser):
+    )
+    subparsers_profile = profile.add_subparsers(
+        description="Query information about buildtest profiles",
+        dest="profile",
```
for consistency make this `profiles`.

kingxleyy replied:
done

shahzebsiddiqui commented on buildtest/cli/config.py:
```diff
@@ -20,9 +20,14 @@ def config_cmd(args, configuration, editor, system):
     if args.config in ["view", "v"]:
         view_configuration(configuration, theme=args.theme, pager=args.pager)
 
+    elif args.config in ["profile"]:
```
same here change this to `profiles`

kingxleyy replied:
done

shahzebsiddiqui commented on buildtest/cli/config.py:
```diff
@@ -20,9 +20,14 @@ def config_cmd(args, configuration, editor, system):
         view_configuration(configuration, theme=args.theme, pager=args.pager)
 
+    elif args.config in ["profile"]:
+        if args.profile == "list":
```
and here to change to 

```python
if args.profiles == "list":
```

kingxleyy replied:
done

shahzebsiddiqui commented on bash_completion.sh:
```diff
@@ -237,7 +237,10 @@ _buildtest ()
-      local cmds="-h --help co compilers e edit ex executors p path systems val validate v view"
+
+
+      local cmds="-h --help co compilers e edit ex executors p path profile systems validate v view"
```
change `profile` -> `profiles`

kingxleyy replied:
done

shahzebsiddiqui commented on bash_completion.sh:
```diff
@@ -269,6 +272,19 @@ _buildtest ()
             return
           esac
         ;;
+        profile)
```
change this to `profiles`

kingxleyy replied:
done

kingxleyy committed:
profile to profiles and added documentation

kingxleyy committed:
profile to profiles and added documentation 2

kingxleyy committed:
removing -t in bash

kingxleyy committed:
removing -t in bash 2

kingxleyy committed:
fixing config. yml

shahzebsiddiqui committed:
applying black

shahzebsiddiqui committed:
removing profiles section from config file
